### PR TITLE
Manage topics from admin page in Target MVD

### PR DESCRIPTION
**Pull request by afmicc**
#### Trello board reference:

* [As an admin I should be able to create/edit/delete new topics for targets](https://trello.com/c/7kfPDPvH)
* [As an admin I should be able to filter targets by topic](https://trello.com/c/xi7UUxGI)

---

#### Description:

* Refactor topic entity from enum to table 
* Add active admin pages to topics

### CODE REVIEW

File: app/models/topic.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#  id         :bigint           not null, primary key
+#  title      :string           not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class Topic < ApplicationRecord
+  has_many :targets, dependent: :destroy
+
+  validates :title, presence: true
```
Comment (gabrielbursztein2) [inline app/models/topic.rb]: Validate uniqueness.

File: db/migrate/20190807202020_create_topics.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateTopics < ActiveRecord::Migration[5.2]
+  def change
+    create_table :topics do |t|
+      t.string :title, null: false
+
+      t.timestamps
+    end
+
+    %w[football travel politics art dating music movies series food].each do |topic|
```
Comment (gabrielbursztein2) [inline db/migrate/20190807202020_create_topics.rb]: This should be in a separate task

File: spec/support/shared_contexts/near_targets.rb [hunk 3]
```diff
@@ -3,12 +3,16 @@
...
+  let!(:topic1) { create(:topic) }
+  let!(:topic2) { create(:topic) }
+
+  # target - Montenvideo Shopping - 7km - compatible: Y
   let!(:target1) do
     create(:target,
            user: user2,
            title: 'Montenvideo Shopping',
-           topic: Target.topics.key(1),
+           topic_id: topic1.id,
```
Comment (gabrielbursztein2) [inline spec/support/shared_contexts/near_targets.rb]: `topic: topic1`

**Review by gabrielbursztein2 — APPROVED** [review]

File: spec/request/api/v1/topics/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    it 'is expected that response contains at least one' do
+      topics = json_value(response.parsed_body, 'topics')
+      expect(topics.count).to be >= 1
+    end
+
+    it 'is expected that response contains at least some body data' do
+      expect(response.body).to include_json(
+        topics: a_collection_including(
+          a_hash_including('id' => topic.id, 'title' => topic.title)
```
Comment (JuanfraM) [inline spec/request/api/v1/topics/index_spec.rb]: I think you can use the new syntaxis here: `id: topic.id`, etc

**Review by JuanfraM — APPROVED** [review]
👌

**Pull request merged**